components/scream/CMakeLists.txt

```diff
@@ -250,6 +247,28 @@ endif()
 
 set (SCREAM_DATA_DIR ${SCREAM_INPUT_ROOT}/atm/scream CACHE PATH "" FORCE)
 
+#
+# Handle test level
+#
+
+# Constants
+set(SCREAM_TEST_LEVEL_AT "0")
+set(SCREAM_TEST_LEVEL_NIGHTLY "1")
+set(SCREAM_TEST_LEVEL_EXPERIMENTAL "2")
+
+set(SCREAM_TEST_LEVEL "AT" CACHE STRING "The test level to run. Default is AT. NIGHTLY will run additional tests but is not guaranteed to PASS. EXPERIMENTAL will run even more tests with failures being more likely")
+
+if (SCREAM_TEST_LEVEL STREQUAL "AT")
```

Comment:
This might be personal taste, but I think that a "test level" suggests that the different values add an increasingly higher amount of testing. This is the case here, but it might be hard for users to know which level is higher. Also, "AT" might not be very self-explanatory for users.

What about
- "FAST" instead of "AT": immediately hints to the fact that tests are relatively quick.
- "FULL", or "STANDARD", or "NORMAL" instead "NIGHTLY"
- "EXTENSIVE" instead of  "EXPERIMENTAL": this should hint to the fact that it covers everything, and might be quite costly.

Reply:
@bartgol , the intent of the SCREAM_TEST_LEVEL is more geared towards how crucial it is that the tests always passes rather than how fast it runs. While it is true that we generally want AT tests to be pretty fast, it would not be unreasonable to put some slower tests at the AT level if that's what we needed to ensure PRs did not break important functionality:

AT: this test must always pass
Nightly: this test should always pass. If it doesn't than we see it on our dashboard and we're working on it.
Experimental: no guarantees. Useful for future/experimental features